src/utils/daos_hdlr.c

```diff
@@ -203,20 +473,343 @@ pool_query_hdlr(struct cmd_args_s *ap)
  * int cont_FN_hdlr(struct cmd_args_s *ap)
  *
  * cont_list_objs_hdlr()
- * int cont_query_hdlr()
  * int cont_stat_hdlr()
- * int cont_get_prop_hdlr()
  * int cont_set_prop_hdlr()
- * int cont_list_attrs_hdlr()
  * int cont_del_attr_hdlr()
- * int cont_get_attr_hdlr()
- * int cont_set_attr_hdlr()
- * int cont_create_snap_hdlr()
- * int cont_list_snaps_hdlr()
- * int cont_destroy_snap_hdlr()
  * int cont_rollback_hdlr()
  */
 
+int
+cont_list_snaps_hdlr(struct cmd_args_s *ap)
+{
+	daos_epoch_t *buf;
+	daos_anchor_t anchor;
+	int rc, i, snaps_count, expected_count;
+
+	/* evaluate size for listing */
+	snaps_count = 0;
+	memset(&anchor, 0, sizeof(anchor));
+	rc = daos_cont_list_snap(ap->cont, &snaps_count, NULL, NULL, &anchor,
+				 NULL);
+	if (rc != 0) {
+		fprintf(stderr, "cont list snaps failed: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+	D_PRINT("Container's snapshots :\n");
+	if (daos_anchor_is_eof < 0) {
+		fprintf(stderr, "invalid number of snapshots returned\n");
+		D_GOTO(out, rc = -DER_INVAL);
+	}
+	if (snaps_count == 0) {
+		D_PRINT("no snapshots\n");
+		D_GOTO(out, rc);
+	}
+
+	D_ALLOC_ARRAY(buf, snaps_count);
+	if (buf == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+
+	expected_count = snaps_count;
+	memset(&anchor, 0, sizeof(anchor));
+	rc = daos_cont_list_snap(ap->cont, &snaps_count, buf, NULL, &anchor,
+				 NULL);
+	if (rc != 0) {
+		fprintf(stderr, "cont list snaps failed: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+	if (expected_count < snaps_count)
+		fprintf(stderr, "size required to gather all snapshots has raised, list has been truncated\n");
+
+	for (i = 0; i < min(expected_count, snaps_count); i++)
+		D_PRINT(DF_U64" ", buf[i]);
+	D_PRINT("\n");
+
+out:
+	return rc;
+}
+
+int
+cont_create_snap_hdlr(struct cmd_args_s *ap)
+{
+	int rc;
+
+	rc = daos_cont_create_snap(ap->cont, &ap->epc, NULL, NULL);
```

Comment:
ap->snapname_str may be NULL or may contain a name, either way it seems it could be provided as the name argument to the API call here.

Reply:
hmmm, if a snapname is set/used it will cause a DER_NOSYS to be returned by API... but ok, will be in next commit